realm/realm-library/src/androidTest/kotlin/io/realm/EmbeddedObjectsTest.kt

```diff
@@ -527,21 +528,155 @@ class EmbeddedObjectsTest {
+                    .size
+                    .also { oldChildCount ->
+                        assertEquals(0, oldChildCount)
+                    }
+        }
     }
 
     @Test
-    @Ignore("Add in another PR")
     fun insert_listWithEmbeddedObjects() {
-        TODO()
+        realm.executeTransaction { realm ->
+            val list = listOf<EmbeddedSimpleParent>()
+                    .plus(EmbeddedSimpleParent("parent1").apply { child = EmbeddedSimpleChild("child1") })
+                    .plus(EmbeddedSimpleParent("parent2").apply { child = EmbeddedSimpleChild("child2") })
+                    .plus(EmbeddedSimpleParent("parent3").apply { child = EmbeddedSimpleChild("child3") })
+                    .also { realm.insert(it) }
+
+            realm.where<EmbeddedSimpleParent>()
+                    .findAll()
+                    .sort("_id")
+                    .also {
+                        assertEquals(3, it.count())
+                        assertEquals(list[0]._id, it[0]!!._id)
+                        assertEquals(list[1]._id, it[1]!!._id)
+                        assertEquals(list[2]._id, it[2]!!._id)
+                        assertEquals(list[0].child!!.childId, it[0]!!.child!!.childId)
+                        assertEquals(list[1].child!!.childId, it[1]!!.child!!.childId)
+                        assertEquals(list[2].child!!.childId, it[2]!!.child!!.childId)
+                    }
+
+            realm.where<EmbeddedSimpleChild>()
+                    .findAll()
+                    .sort("childId")
+                    .also {
+                        assertEquals(3, it.count())
+                        assertEquals(list[0].child!!.childId, it[0]!!.childId)
+                        assertEquals(list[1].child!!.childId, it[1]!!.childId)
+                        assertEquals(list[2].child!!.childId, it[2]!!.childId)
+                        assertEquals(list[0]._id, it[0]!!.parent._id)
+                        assertEquals(list[1]._id, it[1]!!.parent._id)
+                        assertEquals(list[2]._id, it[2]!!.parent._id)
+                    }
+        }
+    }
+
+    @Test
+    fun insert_listWithEmbeddedObjects_duplicatePrimaryKeyThrows() {
+        realm.executeTransaction { realm ->
+            val list = listOf<EmbeddedSimpleParent>()
+                    .plus(EmbeddedSimpleParent("parent1").apply { child = EmbeddedSimpleChild("child1") })
+                    .plus(EmbeddedSimpleParent("parent2").apply { child = EmbeddedSimpleChild("child2") })
+                    .plus(EmbeddedSimpleParent("parent3").apply { child = EmbeddedSimpleChild("child3") })
+                    .also { realm.insert(it) }
+
+            assertFailsWith<RealmPrimaryKeyConstraintException> {
+                realm.insert(list)
+            }
+        }
+    }
+
+    @Test
+    fun insert_listWithEmbeddedObjects_insertingChildrenDirectlyThrows() {
+        realm.executeTransaction { realm ->
+            listOf<EmbeddedSimpleChild>()
```

Comment:
See combination of other comments. Why not just write
```
assertFailsWith<IllegalArgumentException> {
    realm.insert(listOf(....))
}
```

Reply:
Hm. I guess here it actually makes sense, as we only want to expect failures from the insert.